src/Arcus.Messaging.Pumps.Abstractions/Extensions/IMessagePumpCircuitBreakerExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Threading.Tasks;
+using GuardNet;
+
+// ReSharper disable once CheckNamespace
+namespace Arcus.Messaging.Pumps.Abstractions.Transient
+{
+    /// <summary>
+    /// Extensions on the <see cref="IMessagePumpCircuitBreaker"/> for more dev-friendly interaction.
+    /// </summary>
+    // ReSharper disable once InconsistentNaming
+    public static class IMessagePumpCircuitBreakerExtensions
```

Comment:
Yes, that's true, but it felt a bit wrong to 'ruin' the interface with an overload. Adding an extension for this seemed a bit 'cleaner' in that regard. Was also thinking about in the future where it may be easier to change an extension than an interface.
It will make implementations of the interface also a bit easier.

But happy to discus this, not sure what the common practice is here.

Reply:
I'm not sure either.
You have a point in saying that it is an additional method that has been added for improving the developer experience, and it might not be an actual part of the 'contract'.
On the other hand, by adding it to the contract we force implementors to add that method as well but I believe the implementation will be the same for all implementations .... 

So actually, I'm fine with having it as an extension method.   Moreover, I like this point of view.   Good job in having it done this way, but I also think it was worth the effort to have a little contemplation on it.   I think I would 've done it differently (make it a member), but in hindsight, I think the extension method approach is better.

Reply:
Great! Thx for this. Yes, always happy to discus this and future stuff related to dev-experience 👍.

Reply:
Resolving.